### ISSUE TITLE
Add fsi to dotnet watch

### ISSUE THREAD

**Issue by olivercoad** [dotnet/sdk#20907]
`dotnet watch` is great for running a project but doesn't currently support running an f# script.

I propose making the built in `dotnet-watch` tool support fsi.
```
dotnet watch fsi myscript.fsx
```
It can find discover files by recursively looking at `#load` lines, starting from the provided script file.

### Context
Alternatively something like this could be a 3rd party nuget package that could be installed as a dotnet tool.
I have a proof of concept that I made with the intention to publish it as a dotnet tool but ended up copying a lot of the internals of `dotnet-watch` and didn't need to add much, which is why I'm proposing just adding it to the built-in tool.

I would be willing to implement this feature if it gets approved-in-principle.

**Comment by olivercoad** [comment]
@pranavkm is this something likely to be accepted?

**Comment by pranavkm** [comment]
@olivercoad `dotnet watch` will run any `dotnet` command in a loop (e.g. `dotnet watch test` runs `dotnet test` while watching for file changes). I'm not sure if that helps your scenario.  We wouldn't want to expand it beyond this to support arbitrary commands since that dramatically increases our support burden.

**Comment by olivercoad** [comment]
@pranavkm Yes, the only reason it doesn't work as-is is that it doesn't get the fileset.

I think the only thing that would need to change is to implement and use a different `IFileSetFactory` with `DotNetWatcher` and not use `MSBuildEvaluationFilter` when the command is `fsi`.
